packages/blocks/src/api/registration.js

```diff
@@ -232,6 +232,30 @@ export function registerBlockType( blockNameOrMetadata, settings ) {
 		return;
 	}
 
+	// Parent being an array hasn't been enforced in the past,
+	// so this is a way to maintain backwards compatibility
+	// with 3rd party blocks that may have been using it as a string.
+	if (
+		typeof settings?.parent === 'string' ||
+		settings?.parent instanceof String
+	) {
+		settings.parent = [ settings.parent ];
+		warning(
```

Comment:
Actually, I had to remove it at https://github.com/WordPress/gutenberg/pull/66250/commits/32c9a1f62805b1c46b3d3e928810cc0181e26aa4 for tests to pass.

Reply:
Can we amend the tests to expect the warnings? Seems like a better outcome.

Reply:
Yeah, this is better. I was also reluctant to add a warning but given this can have [side-effects](https://github.com/WordPress/gutenberg/pull/66250#discussion_r1810568607), it's best to have it.